Automattic/jetpack
# Add Video Tracks Support to VideoPress Block

Adds support for uploading .vtt files to add captions/subtitles to a VideoPress video. Because of the way VideoPress is embedded with iframes, we need to host and serve the .vtt files from wp.com to avoid CORS issues.

When a track is selected, it will get uploaded to wp.com and attached to the video. The video player will fetch the tracks data from the API video endpoint and load the tracks into the UI. Most of this code is from the core video block's tracks component, with some tweaks added to support the management of the tracks via the API.

**Screenshot**:
<img width="634" alt="Screen Shot 2021-10-29 at 3 32 52 PM" src="https://user-images.githubusercontent.com/789137/139504139-9a8537f2-d80f-4d3c-ad20-6c34430f100b.png">

#### Testing instructions:
 * Add a new video (or choose an old one) to the block editor, click on the captions button, and upload a new track. Only .vtt files should be allowed to be selected and uploaded. 
 * You can add a type, source language and label. Leaving them blank should default to `English`, `en`, and `subtitles`.
 * The track should upload and you should return to the main list.
 * Publish the post and watch the video. Your track(s) should be selectable in the player.
 * Try deleting tracks, they should be removed from the list and no longer show as an option if you refresh the player.
 * When uploading a track, try violating one of the form field requirements, for example the `Source language` field can't be larger than 5 characters. You should see an error message at the bottom of the component.
 * Add the same video to another block, it should fetch the track data form the api and you should see them in the tracks list.
 * Take note of the url for a video track (Look for <track> tags when you view source) If you delete a video from the media library, the tracks files should also be deleted and 404.

#### Does this pull request change what data or activity we track or use?
No.
